lib/modules/versioning/pvp/index.ts

```diff
@@ -0,0 +1,297 @@
+import { logger } from '../../../logger';
+import type { RangeStrategy } from '../../../types/versioning';
+import { regEx } from '../../../util/regex';
+import type { NewValueConfig, VersioningApi } from '../types';
+
+export const id = 'pvp';
+export const displayName = 'Package Versioning Policy (Haskell)';
+export const urls = [];
+export const supportsRanges = true;
+export const supportedRangeStrategies: RangeStrategy[] = ['auto'];
+
+type Parsed = { lower: string; upper: string };
+type Components = { major: number[]; minor: number[]; patch: number[] };
+
+export function parse(input: string): Parsed | null {
+  const noSpaces = input.replaceAll(' ', '');
+  const r = regEx(/>=(?<lower>[\d.]+)&&<(?<upper>[\d.]+)/);
+  const m = r.exec(noSpaces);
+  if (!m?.groups) {
+    return null;
+  }
+  return {
+    lower: m.groups['lower'],
+    upper: m.groups['upper'],
+  };
+}
+
+export function extractAllComponents(version: string): number[] {
+  const versionMajor = version.split('.');
+  const versionIntMajor: number[] = versionMajor.map((x) => parseInt(x, 10));
```

Comment:
```suggestion
  const versionIntMajor = versionMajor.map(parseInt);
```

Reply:
Your suggested code won't do the right thing, see https://stackoverflow.com/a/262511/309483 . Also, I think it is nice to explicitly state the base, since parseInt will happily parse hex numbers if the base is not specified. That should never happen, but we might as well rule it out.

```
> parseInt('0x10')
16
```

I omit the type as suggested.